test/export/StructureDefinition.ResourceExporter.test.ts

```diff
@@ -269,12 +269,86 @@ describe('ResourceExporter', () => {
-    expect(exported.elements).toHaveLength(12); // 9 AlternateIdentification elements + 3 added elements
+  it('should log an error when adding an element with the same path as an inherited element', () => {
+    const resource = new Resource('MyResource');
+    const addElementRule = new AddElementRule('extension');
+    addElementRule.min = 0;
+    addElementRule.max = '1';
+    resource.rules.push(addElementRule);
+    doc.resources.set(resource.name, resource);
+    const exported = exporter.export().resources;
+
+    expect(exported.length).toBe(1);
+    expect(exported[0].elements.filter(r => r.id === 'MyResource.extension')).toHaveLength(1);
+
+    expect(exported[0].name).toBe('MyResource');
+    expect(exported[0].id).toBe('MyResource');
+    expect(exported[0].type).toBe('MyResource');
+    expect(exported[0].baseDefinition).toBe(
+      'http://hl7.org/fhir/StructureDefinition/DomainResource'
+    );
+    expect(exported[0].elements).toHaveLength(9);
+
+    expect(loggerSpy.getAllMessages('error')).toHaveLength(1);
+    expect(loggerSpy.getLastMessage('error')).toMatch(
+      `Cannot define element ${addElementRule.path} on ${resource.name} because it has already been defined`
+    );
+  });
+
+  it('should log an error when two rules add a new element with the same path', () => {
+    const resource = new Resource('MyResource');
+    const addElementRule = new AddElementRule('testElement');
+    addElementRule.min = 0;
+    addElementRule.max = '1';
+    const addElementRule2 = new AddElementRule('testElement');
+    addElementRule2.min = 0;
+    addElementRule2.max = '1';
+    resource.rules.push(addElementRule);
+    resource.rules.push(addElementRule2);
+    doc.resources.set(resource.name, resource);
+    const exported = exporter.export().resources;
+
+    expect(exported.length).toBe(1);
+    expect(exported[0].elements.filter(r => r.id === 'MyResource.testElement')).toHaveLength(1);
+
+    expect(exported[0].name).toBe('MyResource');
+    expect(exported[0].id).toBe('MyResource');
+    expect(exported[0].type).toBe('MyResource');
+    expect(exported[0].baseDefinition).toBe(
+      'http://hl7.org/fhir/StructureDefinition/DomainResource'
+    );
+    expect(exported[0].elements).toHaveLength(10);
+
+    expect(loggerSpy.getAllMessages('error')).toHaveLength(1);
+    expect(loggerSpy.getLastMessage('error')).toMatch(
+      `Cannot define element ${addElementRule.path} on ${resource.name} because it has already been defined`
+    );
+  });
+
+  it('should log an error when a rule with the same path is added by directly calling newElement', () => {
+    doc = new FSHDocument('fileName');
+    const input = new FSHTank([doc], minimalConfig);
+    pkg = new Package(input.config);
+    const fisher = new TestFisher(input, defs, pkg);
+    const alternateIdentification: StructureDefinition =
+      fisher.fishForStructureDefinition('AlternateIdentification');
```

Comment:
I'm seeing a type error on this line. Since the `fishForStructureDefinition` returns `StructureDefinition | undefined`, it doesn't like that `alternateIdentification` has a type of `StructureDefinition`. In this case, we know that `AlternateIdentification` can be fished, so you can add `as StructureDefinition` to say you're sure it will return that type, rather than `undefined`. 


```suggestion
    const alternateIdentification: StructureDefinition =
      fisher.fishForStructureDefinition('AlternateIdentification') as StructureDefinition;
```

Reply:
Done!